src/openrct2-ui/windows/ScenarioSelect.cpp

```diff
@@ -82,76 +81,61 @@ enum {
+    MakeRemapWidget({ 3, TabsStart + (TabHeight * 5) }, { TabWidth, TabHeight}, WindowWidgetType::Tab, WindowColour::Secondary, SPR_G2_SIDEWAYS_TAB), // tab 06
+    MakeRemapWidget({ 3, TabsStart + (TabHeight * 6) }, { TabWidth, TabHeight}, WindowWidgetType::Tab, WindowColour::Secondary, SPR_G2_SIDEWAYS_TAB), // tab 07
+    MakeRemapWidget({ 3, TabsStart + (TabHeight * 7) }, { TabWidth, TabHeight}, WindowWidgetType::Tab, WindowColour::Secondary, SPR_G2_SIDEWAYS_TAB), // tab 08
+    MakeRemapWidget({ 3, TabsStart + (TabHeight * 8) }, { TabWidth, TabHeight}, WindowWidgetType::Tab, WindowColour::Secondary, SPR_G2_SIDEWAYS_TAB), // tab 09
+    MakeRemapWidget({ 3, TabsStart + (TabHeight * 8) }, { TabWidth, TabHeight}, WindowWidgetType::Tab, WindowColour::Secondary, SPR_G2_SIDEWAYS_TAB), // tab 10
+    MakeWidget({ TabWidth + 3, WidgetsStart + 1 }, { WW - SidebarWidth, 276 }, WindowWidgetType::Scroll, WindowColour::Secondary, SCROLL_VERTICAL), // level list
     WIDGETS_END,
 };
+// clang-format on
 
-static constexpr const StringId ScenarioOriginStringIds[] = {
-    STR_SCENARIO_CATEGORY_RCT1,
-    STR_SCENARIO_CATEGORY_RCT1_AA,
-    STR_SCENARIO_CATEGORY_RCT1_LL,
-    STR_SCENARIO_CATEGORY_RCT2,
-    STR_SCENARIO_CATEGORY_RCT2_WW,
-    STR_SCENARIO_CATEGORY_RCT2_TT,
-    STR_SCENARIO_CATEGORY_UCES,
-    STR_SCENARIO_CATEGORY_REAL_PARKS,
-    STR_SCENARIO_CATEGORY_EXTRAS_PARKS,
-    STR_SCENARIO_CATEGORY_OTHER_PARKS,
-};
-
-static void WindowScenarioselectInitTabs(WindowBase *w);
+static void WindowScenarioselectInitTabs(WindowBase* w);
+static void DrawCategoryHeading(WindowBase* w, DrawPixelInfo* dpi, int32_t left, int32_t right, int32_t y, StringId stringId);
+static void InitialiseListItems(WindowBase* w);
+static bool IsScenarioVisible(WindowBase* w, const ScenarioIndexEntry* scenario);
+static bool IsLockingEnabled(WindowBase* w);
 
-static void WindowScenarioselectClose(WindowBase *w);
-static void WindowScenarioselectMouseup(WindowBase *w, WidgetIndex widgetIndex);
-static void WindowScenarioselectMousedown(WindowBase *w, WidgetIndex widgetIndex, Widget* widget);
-static void WindowScenarioselectScrollgetsize(WindowBase *w, int32_t scrollIndex, int32_t *width, int32_t *height);
-static void WindowScenarioselectScrollmousedown(WindowBase *w, int32_t scrollIndex, const ScreenCoordsXY& screenCoords);
-static void WindowScenarioselectScrollmouseover(WindowBase *w, int32_t scrollIndex, const ScreenCoordsXY& screenCoords);
-static void WindowScenarioselectInvalidate(WindowBase *w);
-static void WindowScenarioselectPaint(WindowBase *w, DrawPixelInfo *dpi);
-static void WindowScenarioselectScrollpaint(WindowBase *w, DrawPixelInfo *dpi, int32_t scrollIndex);
+static bool _titleEditor = false;
+static bool _disableLocking{};
+static std::function<void(std::string_view)> _callback;
+static std::vector<ScenarioListItem> _listItems;
```

Comment:
These can all be static members (except maybe for callback - will need to investigate that one further).